csds-client/client/client.go

```diff
@@ -0,0 +1,255 @@
+package client
+
+import (
+	csdspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v2"
+	envoy_type_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+
+	"context"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/metadata"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+type Flag struct {
+	uri         string
+	platform    string
+	authnMode   string
+	apiVersion  string
+	requestFile string
+	requestYaml string
+	jwt         string
+	configFile  string
+	monitorFreq string
+}
+
+type Client struct {
+	cc         *grpc.ClientConn
+	csdsClient csdspb.ClientStatusDiscoveryServiceClient
+
+	nm   []*envoy_type_matcher.NodeMatcher
+	md   metadata.MD
+	info Flag
+}
+
+// ParseFLags parses flags to info
+func ParseFlags() Flag {
+	uriPtr := flag.String("service_uri", "trafficdirector.googleapis.com:443", "the uri of the service to connect to")
+	platformPtr := flag.String("cloud_platform", "gcp", "the cloud platform (e.g. gcp, aws,  ...)")
+	authnModePtr := flag.String("authn_mode", "auto", "the method to use for authentication (e.g. auto, jwt, ...)")
+	apiVersionPtr := flag.String("api_version", "v2", "which xds api major version  to use (e.g. v2, v3 ...)")
+	requestFilePtr := flag.String("request_file", "", "yaml file that defines the csds request")
+	requestYamlPtr := flag.String("request_yaml", "", "yaml string that defines the csds request")
+	jwtPtr := flag.String("jwt_file", "", "path of the -jwt_file")
+	configFilePtr := flag.String("file_to_save_config", "", "the file name to save config")
+	monitorFreqPtr := flag.String("monitor_freq", "", "the frequency of sending request in monitor mode (e.g. 500ms, 2s, 1m ...)")
```

Comment:
Why not duration flag? https://golang.org/pkg/flag/#Duration

Reply:
Make sense. There's no need for parsing the string to duration later. Changed the type of this flag to Duration.